pilot/spec/build_manager_spec.rb

```diff
@@ -221,4 +223,61 @@
       end
     end
   end
+
+  describe "#upload" do
+    describe "spaceship login (via Manager.login)" do
+      let(:fake_build_manager) { Pilot::BuildManager.new }
+      let(:upload_options) do
+        {
+          apple_id: 'mock_apple_id',
+          skip_waiting_for_build_processing: true,
+          ipa: 'foo'
+        }
+      end
+
+      before(:each) do
+        allow(fake_build_manager).to receive(:fetch_app_platform).and_return('ios')
+
+        fake_ipauploadpackagebuilder = double
+        allow(fake_ipauploadpackagebuilder).to receive(:generate).and_return(true)
+        allow(FastlaneCore::IpaUploadPackageBuilder).to receive(:new).and_return(fake_ipauploadpackagebuilder)
+
+        fake_itunestransporter = double
+        allow(fake_itunestransporter).to receive(:upload).and_return(true)
+        allow(FastlaneCore::ItunesTransporter).to receive(:new).and_return(fake_itunestransporter)
+      end
+
+      it "is not executed when skip_waiting_for_build_processing and apple_id are set" do
+        fake_build_manager.upload(upload_options)
```

Comment:
I am unclear on how we assert that `upload` is not executed here.

Reply:
This test is: When using the upload, no login is executed, under special circumstances.
Although this is only verified by all the other stuff from line 254 and following not being present.

Reply:
Taken care of with the `expect.not to.`